docs/descriptor.rst

```diff
@@ -99,6 +99,10 @@ are on the following form::
 
     sgtk:descriptor:app_store?name=tk-core&version=v12.3.4
 
+App store may also have an optional ``label`` parameter. This indicates that the descriptor is tracking
```

Comment:
Feels like we should elaborate here. Specifically not having to specify a `version` when using the `label` param. Would also be good to provide an example similar to the one in the PR description.